### PR TITLE
James 2293 2

### PR DESCRIPTION
**Pull request by rouazana**
Following of https://github.com/linagora/james-project/pull/1245

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java [hunk 1]
```diff
@@ -237,7 +246,60 @@ public void defineDeleteMail() {
...
-private Optional<Integer> assertPositiveInteger(Request request, String parameterName) {
+    @PATCH
+    @Path("/{encodedUrl}/mails")
+    @ApiOperation(value = "Deleting all mails in that mailRepository")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = true,
+            paramType = "query parameter",
+            dataType = "String",
+            example = "?action=clear",
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java]: ok, I realize that I didn't read the updated spec to the end.
What about `DELETE /mailRepositories/{encodedUrl}/mails` ?

File: server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/WebAdminUtils.java [hunk 2]
```diff
@@ -49,11 +49,15 @@ public static WebAdminServer createWebAdminServer(MetricFactory metricFactory, R
     }
 
     public static RequestSpecBuilder defineRequestSpecification(WebAdminServer webAdminServer) {
+        return defineRequestSpecification(webAdminServer.getPort());
+    }
+
+    public static RequestSpecBuilder defineRequestSpecification(PortSupplier portSupplier) {
```
Comment (mbaechler) [inline server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/WebAdminUtils.java]: s/build/define ?

File: server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java (hunk 1 above)
Comment (rouazana) [reply server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java]: I also proposed that to @chibenwa and he's telling that that means removing the resource /mails (which is not, it's just emptied), so he prefers this. WDYT?

Comment (mbaechler) [reply server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java]: That fact is it doesn't DELETE the repository but the mails, that looks ok to me.

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java]: Ok

**Comment by chibenwa** [comment]
```
[98ffa3b82ff5551d44d95ba8c0b7d4d9fc3d54a6] Tests in error: 
[98ffa3b82ff5551d44d95ba8c0b7d4d9fc3d54a6]   DefaultMailboxesProvisioningFilterTest.createMailboxesIfNeededShouldNotGenerateExceptionsInConcurrentEnvironment:83 » Execution
```

**Comment by rouazana** [comment]
👍

**Comment by chibenwa** [comment]
test this please

Heap space

**Comment by chibenwa** [comment]
Heap space again. test this please

**Comment by chibenwa** [comment]
Heap space.

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
🍏

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**